diff -urN gst-ffmpeg-0.10.4-old/ext/ffmpeg/gstffmpeg.c gst-ffmpeg-0.10.4/ext/ffmpeg/gstffmpeg.c
--- gst-ffmpeg-0.10.4-old/ext/ffmpeg/gstffmpeg.c	2008-05-19 21:58:09.000000000 +0200
+++ gst-ffmpeg-0.10.4/ext/ffmpeg/gstffmpeg.c	2008-10-05 01:15:01.000000000 +0200
@@ -31,8 +31,8 @@
 #include <avcodec.h>
 #include <avformat.h>
 #else
-#include <ffmpeg/avcodec.h>
-#include <ffmpeg/avformat.h>
+#include <libavcodec/avcodec.h>
+#include <libavformat/avformat.h>
 #endif
 
 #include "gstffmpeg.h"
diff -urN gst-ffmpeg-0.10.4-old/ext/ffmpeg/gstffmpegcodecmap.c gst-ffmpeg-0.10.4/ext/ffmpeg/gstffmpegcodecmap.c
--- gst-ffmpeg-0.10.4-old/ext/ffmpeg/gstffmpegcodecmap.c	2008-05-19 21:58:09.000000000 +0200
+++ gst-ffmpeg-0.10.4/ext/ffmpeg/gstffmpegcodecmap.c	2008-10-05 01:15:36.000000000 +0200
@@ -27,8 +27,8 @@
 #include <avcodec.h>
 #include <libswscale/swscale.h>
 #else
-#include <ffmpeg/swscale.h>
-#include <ffmpeg/avcodec.h>
+#include <libswscale/swscale.h>
+#include <libavcodec/avcodec.h>
 #endif
 #include <string.h>
 
@@ -332,7 +332,7 @@
           NULL);
       if (context) {
         gst_caps_set_simple (caps,
-            "depth", G_TYPE_INT, context->bits_per_sample, NULL);
+            "depth", G_TYPE_INT, context->bits_per_coded_sample, NULL);
       }
       break;
 
@@ -511,7 +511,7 @@
       caps = gst_ff_vid_caps_new (context, codec_id, "video/x-huffyuv", NULL);
       if (context) {
         gst_caps_set_simple (caps,
-            "bpp", G_TYPE_INT, context->bits_per_sample, NULL);
+            "bpp", G_TYPE_INT, context->bits_per_coded_sample, NULL);
       }
       break;
 
@@ -627,7 +627,7 @@
           "layout", G_TYPE_STRING, "microsoft", NULL);
       if (context) {
         gst_caps_set_simple (caps,
-            "depth", G_TYPE_INT, (gint) context->bits_per_sample, NULL);
+            "depth", G_TYPE_INT, (gint) context->bits_per_coded_sample, NULL);
       } else {
         gst_caps_set_simple (caps, "depth", GST_TYPE_INT_RANGE, 1, 64, NULL);
       }
@@ -638,7 +638,7 @@
           "layout", G_TYPE_STRING, "quicktime", NULL);
       if (context) {
         gst_caps_set_simple (caps,
-            "depth", G_TYPE_INT, (gint) context->bits_per_sample, NULL);
+            "depth", G_TYPE_INT, (gint) context->bits_per_coded_sample, NULL);
       } else {
         gst_caps_set_simple (caps, "depth", GST_TYPE_INT_RANGE, 1, 64, NULL);
       }
@@ -688,7 +688,7 @@
       caps = gst_ff_vid_caps_new (context, codec_id, "video/x-camtasia", NULL);
       if (context) {
         gst_caps_set_simple (caps,
-            "depth", G_TYPE_INT, (gint) context->bits_per_sample, NULL);
+            "depth", G_TYPE_INT, (gint) context->bits_per_coded_sample, NULL);
       } else {
         gst_caps_set_simple (caps, "depth", GST_TYPE_INT_RANGE, 8, 32, NULL);
       }
@@ -1026,7 +1026,7 @@
       caps = gst_ff_aud_caps_new (context, codec_id, "audio/x-alac", NULL);
       if (context) {
         gst_caps_set_simple (caps,
-            "samplesize", G_TYPE_INT, context->bits_per_sample, NULL);
+            "samplesize", G_TYPE_INT, context->bits_per_coded_sample, NULL);
       }
       break;
 
@@ -1049,7 +1049,7 @@
       caps = gst_ff_aud_caps_new (context, codec_id, "audio/x-tta", NULL);
       if (context) {
         gst_caps_set_simple (caps,
-            "samplesize", G_TYPE_INT, context->bits_per_sample, NULL);
+            "samplesize", G_TYPE_INT, context->bits_per_coded_sample, NULL);
       }
       break;
     default:
@@ -1413,7 +1413,7 @@
 
   gst_structure_get_int (structure, "width", &context->width);
   gst_structure_get_int (structure, "height", &context->height);
-  gst_structure_get_int (structure, "bpp", &context->bits_per_sample);
+  gst_structure_get_int (structure, "bpp", &context->bits_per_coded_sample);
 
   fps = gst_structure_get_value (structure, "framerate");
   if (fps != NULL && GST_VALUE_HOLDS_FRACTION (fps)) {
@@ -1644,7 +1644,7 @@
       gint depth;
 
       if (gst_structure_get_int (str, "depth", &depth)) {
-        context->bits_per_sample = depth;
+        context->bits_per_coded_sample = depth;
       } else {
         GST_WARNING ("No depth field in caps %" GST_PTR_FORMAT, caps);
       }
@@ -1677,7 +1677,7 @@
         context->bit_rate = bitrate;
     }
     case CODEC_ID_ALAC:
-      gst_structure_get_int (str, "samplesize", &context->bits_per_sample);
+      gst_structure_get_int (str, "samplesize", &context->bits_per_coded_sample);
       break;
 
     case CODEC_ID_DVVIDEO:
diff -urN gst-ffmpeg-0.10.4-old/ext/ffmpeg/gstffmpegcodecmap.h gst-ffmpeg-0.10.4/ext/ffmpeg/gstffmpegcodecmap.h
--- gst-ffmpeg-0.10.4-old/ext/ffmpeg/gstffmpegcodecmap.h	2007-01-09 15:59:34.000000000 +0100
+++ gst-ffmpeg-0.10.4/ext/ffmpeg/gstffmpegcodecmap.h	2008-10-05 01:15:01.000000000 +0200
@@ -23,7 +23,7 @@
 #ifdef HAVE_FFMPEG_UNINSTALLED
 #include <avcodec.h>
 #else
-#include <ffmpeg/avcodec.h>
+#include <libavcodec/avcodec.h>
 #endif
 #include <gst/gst.h>
 
diff -urN gst-ffmpeg-0.10.4-old/ext/ffmpeg/gstffmpegdemux.c gst-ffmpeg-0.10.4/ext/ffmpeg/gstffmpegdemux.c
--- gst-ffmpeg-0.10.4-old/ext/ffmpeg/gstffmpegdemux.c	2008-05-19 21:58:09.000000000 +0200
+++ gst-ffmpeg-0.10.4/ext/ffmpeg/gstffmpegdemux.c	2008-10-05 01:15:01.000000000 +0200
@@ -30,7 +30,7 @@
 #include <avi.h>
 #endif
 #else
-#include <ffmpeg/avformat.h>
+#include <libavformat/avformat.h>
 #ifdef HAVE_AVI_H
 #include <ffmpeg/avi.h>
 #endif
diff -urN gst-ffmpeg-0.10.4-old/ext/ffmpeg/gstffmpegenc.c gst-ffmpeg-0.10.4/ext/ffmpeg/gstffmpegenc.c
--- gst-ffmpeg-0.10.4-old/ext/ffmpeg/gstffmpegenc.c	2008-05-19 21:58:09.000000000 +0200
+++ gst-ffmpeg-0.10.4/ext/ffmpeg/gstffmpegenc.c	2008-10-05 01:16:11.000000000 +0200
@@ -31,7 +31,7 @@
 #ifdef HAVE_FFMPEG_UNINSTALLED
 #include <avcodec.h>
 #else
-#include <ffmpeg/avcodec.h>
+#include <libavcodec/avcodec.h>
 #endif
 
 #include <gst/gst.h>
diff -urN gst-ffmpeg-0.10.4-old/ext/ffmpeg/gstffmpeg.h gst-ffmpeg-0.10.4/ext/ffmpeg/gstffmpeg.h
--- gst-ffmpeg-0.10.4-old/ext/ffmpeg/gstffmpeg.h	2008-05-19 21:58:09.000000000 +0200
+++ gst-ffmpeg-0.10.4/ext/ffmpeg/gstffmpeg.h	2008-10-05 01:15:01.000000000 +0200
@@ -28,8 +28,8 @@
 #include <avcodec.h>
 #include <avformat.h>
 #else
-#include <ffmpeg/avcodec.h>
-#include <ffmpeg/avformat.h>
+#include <libavcodec/avcodec.h>
+#include <libavformat/avformat.h>
 #endif
 
 #include <gst/gst.h>
diff -urN gst-ffmpeg-0.10.4-old/ext/ffmpeg/gstffmpegmux.c gst-ffmpeg-0.10.4/ext/ffmpeg/gstffmpegmux.c
--- gst-ffmpeg-0.10.4-old/ext/ffmpeg/gstffmpegmux.c	2008-05-19 21:58:10.000000000 +0200
+++ gst-ffmpeg-0.10.4/ext/ffmpeg/gstffmpegmux.c	2008-10-05 01:15:02.000000000 +0200
@@ -25,7 +25,7 @@
 #ifdef HAVE_FFMPEG_UNINSTALLED
 #include <avformat.h>
 #else
-#include <ffmpeg/avformat.h>
+#include <libavformat/avformat.h>
 #endif
 
 #include <gst/gst.h>
diff -urN gst-ffmpeg-0.10.4-old/ext/ffmpeg/gstffmpegprotocol.c gst-ffmpeg-0.10.4/ext/ffmpeg/gstffmpegprotocol.c
--- gst-ffmpeg-0.10.4-old/ext/ffmpeg/gstffmpegprotocol.c	2007-11-03 17:14:53.000000000 +0100
+++ gst-ffmpeg-0.10.4/ext/ffmpeg/gstffmpegprotocol.c	2008-10-05 01:15:02.000000000 +0200
@@ -26,7 +26,7 @@
 #ifdef HAVE_FFMPEG_UNINSTALLED
 #include <avformat.h>
 #else
-#include <ffmpeg/avformat.h>
+#include <libavformat/avformat.h>
 #endif
 
 #include <gst/gst.h>
